### ISSUE TITLE
Compound model crashes (Const1 + Gaussian1)

### ISSUE THREAD

**Issue by hcferguson** [nmearl/pyfocal#87]
Tried to fit a constant + a Gaussian to the region from ~ 1 to 2.2 microns in example_spec_ascii.txt. I selected each model, clicked "Add", edited the parameters, then typed into the "Arithmetic Behavior" box `Const1 + Gaussian1`, then pushed `Create`. It crashed. Traceback below.

```
(pyf_env)Henrys-MacBook-Pro:specview hcferguson$ pyfocal
Traceback (most recent call last):
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/ui/controller.py", line 41, in <lambda>
    lambda: self.new_plot_sub_window())
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/ui/controller.py", line 259, in new_plot_sub_window
    self.add_plot(data=data, window=window)
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/ui/controller.py", line 283, in add_plot
    raise AttributeError("Data is not defined")
AttributeError: Data is not defined
Abort trap: 6
(pyf_env)Henrys-MacBook-Pro:specview hcferguson$ pyfocal
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/astropy/units/quantity.py:822: RuntimeWarning: invalid value encountered in true_divide
  return super(Quantity, self).__truediv__(other)
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/numpy/core/_methods.py:59: RuntimeWarning: Mean of empty slice.
  warnings.warn("Mean of empty slice.", RuntimeWarning)
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/numpy/core/_methods.py:70: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/numpy/core/_methods.py:82: RuntimeWarning: Degrees of freedom <= 0 for slice
  warnings.warn("Degrees of freedom <= 0 for slice", RuntimeWarning)
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/numpy/core/_methods.py:94: RuntimeWarning: invalid value encountered in true_divide
  arrmean, rcount, out=arrmean, casting='unsafe', subok=False)
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/numpy/core/_methods.py:116: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
(pyf_env)Henrys-MacBook-Pro:specview hcferguson$ pyfocal
Traceback (most recent call last):
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/ui/controller.py", line 41, in <lambda>
    lambda: self.new_plot_sub_window())
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/ui/controller.py", line 259, in new_plot_sub_window
    self.add_plot(data=data, window=window)
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/ui/controller.py", line 283, in add_plot
    raise AttributeError("Data is not defined")
AttributeError: Data is not defined
Abort trap: 6
(pyf_env)Henrys-MacBook-Pro:specview hcferguson$ pyfocal
Traceback (most recent call last):
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/ui/controller.py", line 41, in <lambda>
    lambda: self.new_plot_sub_window())
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/ui/controller.py", line 259, in new_plot_sub_window
    self.add_plot(data=data, window=window)
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/ui/controller.py", line 283, in add_plot
    raise AttributeError("Data is not defined")
AttributeError: Data is not defined
Abort trap: 6
(pyf_env)Henrys-MacBook-Pro:specview hcferguson$ pyfocal
Traceback (most recent call last):
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/ui/controller.py", line 41, in <lambda>
    lambda: self.new_plot_sub_window())
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/ui/controller.py", line 259, in new_plot_sub_window
    self.add_plot(data=data, window=window)
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/ui/controller.py", line 283, in add_plot
    raise AttributeError("Data is not defined")
AttributeError: Data is not defined
Abort trap: 6
(pyf_env)Henrys-MacBook-Pro:specview hcferguson$ pyfocal
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/astropy/units/quantity.py:822: RuntimeWarning: invalid value encountered in true_divide
  return super(Quantity, self).__truediv__(other)
(pyf_env)Henrys-MacBook-Pro:specview hcferguson$ pyfocal
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/astropy/units/quantity.py:822: RuntimeWarning: invalid value encountered in true_divide
  return super(Quantity, self).__truediv__(other)
WARNING: Model is linear in parameters; consider using linear fitting methods. [astropy.modeling.fitting]
WARNING:astropy:Model is linear in parameters; consider using linear fitting methods.
WARNING:root:Both actual and predicted relative reductions in the sum of squares
  are at most 0.000000 and the relative error between two consecutive iterates is at 
  most 0.000000
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/astropy/units/quantity.py:822: RuntimeWarning: invalid value encountered in true_divide
  return super(Quantity, self).__truediv__(other)
(pyf_env)Henrys-MacBook-Pro:specview hcferguson$ pyfocal
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/astropy/units/quantity.py:822: RuntimeWarning: invalid value encountered in true_divide
  return super(Quantity, self).__truediv__(other)
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/numpy/core/_methods.py:59: RuntimeWarning: Mean of empty slice.
  warnings.warn("Mean of empty slice.", RuntimeWarning)
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/numpy/core/_methods.py:70: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/numpy/core/_methods.py:82: RuntimeWarning: Degrees of freedom <= 0 for slice
  warnings.warn("Degrees of freedom <= 0 for slice", RuntimeWarning)
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/numpy/core/_methods.py:94: RuntimeWarning: invalid value encountered in true_divide
  arrmean, rcount, out=arrmean, casting='unsafe', subok=False)
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/numpy/core/_methods.py:116: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
Traceback (most recent call last):
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/ui/controller.py", line 342, in update_model_layer
    self.viewer.current_model_formula)
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/interfaces/managers.py", line 221, in update_model
    model = self.get_compound_model(model_dict, formula)
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/interfaces/managers.py", line 210, in get_compound_model
    return self._evaluate(models, formula)
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/interfaces/managers.py", line 164, in _evaluate
    dict(pair for pair in zip(vars, sorted_models)))
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/third_party/py_expression_eval/__init__.py", line 587, in evaluate
    return self.parse(expr).evaluate(variables)
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/third_party/py_expression_eval/__init__.py", line 128, in evaluate
    raise Exception('undefined variable: ' + item.index_)
Exception: undefined variable: Const1
Abort trap: 6
(pyf_env)Henrys-MacBook-Pro:specview hcferguson$ pyfocal
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/astropy/units/quantity.py:822: RuntimeWarning: invalid value encountered in true_divide
  return super(Quantity, self).__truediv__(other)
Traceback (most recent call last):
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/ui/controller.py", line 306, in new_model_layer
    formula=self.viewer.current_model_formula)
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/interfaces/managers.py", line 210, in get_compound_model
    return self._evaluate(models, formula)
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/interfaces/managers.py", line 164, in _evaluate
    dict(pair for pair in zip(vars, sorted_models)))
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/third_party/py_expression_eval/__init__.py", line 587, in evaluate
    return self.parse(expr).evaluate(variables)
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/third_party/py_expression_eval/__init__.py", line 128, in evaluate
    raise Exception('undefined variable: ' + item.index_)
Exception: undefined variable: Const1
Abort trap: 6
```

**Comment by hcferguson** [comment]
It works when I use the following workflow:

 - Select `Const`, click `Add`, edit the parameter, push `Create`
 - Select `Gaussian`, click `Add`, edit the parameters, push `Update` (not entering anything into the `Arithmetic behavior` box.

However, even though I gave it pretty good first guesses, the fit went off in the weeds. It worked okay when I gave it very good initial guesses. (This is general finicky-ness of the astropy optimizers I think...).
